packages/components/timed-text-editor/UpdateTimestamps/index.js

```diff
@@ -48,41 +44,31 @@ const createContentFromEntityList = (currentContent, newEntities) => {
       data: {
         speaker: speaker,
         words: blockEntites,
-        start: blockEntites[0].start
+        start: blockEntites[0]? blockEntites[0].start : 0
       },
-      entityRanges: generateEntitiesRanges(blockEntites, 'punct'),
+      entityRanges: []
     };
 
     updatedBlockArray.push(updatedBlock);
     totalWords += wordsInBlock;
   }
 
-  return { blocks: updatedBlockArray, entityMap: createEntityMap(updatedBlockArray) };
+  return { blocks: updatedBlockArray, entityMap: {}};
 };
 
 // Update timestamps usign stt-align (bbc).
+// depWordsWithCharOffset
 const updateTimestamps = (currentContent, originalContent) => {
   const currentText = convertContentToText(currentContent);
-
-  const entityMap = originalContent.entityMap;
-
-  const entities = [];
-
-  for (const entityIdx in entityMap) {
-    entities.push({
-      start: parseFloat(entityMap[entityIdx].data.start),
-      end: parseFloat(entityMap[entityIdx].data.end),
-      word: entityMap[entityIdx].data.text,
-    });
-  }
-
+  const entities  = originalContent.map((word)=>{
+    word.word = word.text;
+    return word;
+  });
```

Comment:
There's some funky behaviour when editing objects within a list (see at the bottom for example) - it seems to mutate the original too. What you need is a [deep copy](https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript).
```suggestion
  const entities = JSON.parse(JSON.stringify(originalContent));
  entities.map((word)=>{
    word.word = word.text;
    return word;
  });
```
Example of mutation:
```js
> a
[
  { a: 'asdf', b: 1, c: 1, d: 1, e: 1, f: 1 },
  { a: 'asdf', b: 3, c: 2, d: 2, e: 2, f: 2 }
]

> let c = a.map(word => {
... word.b = "asdf"
... return word
... }
... )
>
> c
[
  { a: 'asdf', b: 'asdf', c: 1, d: 1, e: 1, f: 1 },
  { a: 'asdf', b: 'asdf', c: 2, d: 2, e: 2, f: 2 }
]
> a
[
  { a: 'asdf', b: 'asdf', c: 1, d: 1, e: 1, f: 1 },
  { a: 'asdf', b: 'asdf', c: 2, d: 2, e: 2, f: 2 }
]